src/lua/api-gateway/validation/oauth2/userProfileValidator.lua

```diff
@@ -188,10 +188,9 @@ function _M:validateUserProfile()
 
             local cachingObj = self:extractContextVars(json)
 
-            self:setContextProperties(self:getContextPropertiesObject(cachingObj))
-            self:storeProfileInCache(cacheLookupKey, cachingObj)
-
             if ( self:isProfileValid(cachingObj) == true ) then
+                self:setContextProperties(self:getContextPropertiesObject(cachingObj))
```

Comment:
there's a nuance that a profile may be _valid_ for a service, while that same profile may be _invalid_ for another service ( due to scopes, for instance ). I'm not saying it's not right to cache it only when it's valid, just wanted to add to the picture.